### ISSUE TITLE
Getting issue when i deploy to docker registry

### ISSUE THREAD

**Issue by kalanchiyamtutorial** [microsoft/vscode-docker#1959]
Does this occur consistently? 
Repro steps:

1.
2.

Action: vscode-docker.registries.logInToDockerCli
Error type: Error
Error Message: In order to log in to the Docker CLI using tokens, you currently need to go to 
Open your Docker config file and remove "credsStore": "wincred" from the config.json file, then try again. 
Doing this will disable wincred and cause Docker to store credentials directly in the .docker/config.json file. All registries that are currently logged in will be effectly logged out.

Version: 1.1.0
OS: win32
OS Release: 10.0.17763
Product: Visual Studio Code
Product Version: 1.44.2
Language: en

<details>
<summary>Call Stack</summary>

```
extension.bundle.js:164:252001extension.bundle.js:164:252001
ChildProcess.exithandler child_process.js:315:5
ChildProcess.emit events.js:203:13
maybeClose child_process.js:1021:16
Process.ChildProcess._handle.onexit child_process.js:283:5
```

</details>

**Comment by bwateratmsft** [comment]
Have you tried removing `"credsStore": "wincred"` from the Docker config.json?

**Comment by Naninani** [comment]
> Have you tried removing `"credsStore": "wincred"` from the Docker config.json?

Removing the `"credsStore": "wincred"` is exposing the auth token in the config file. Sure that works, but how is that considered secure and compliant approach?

**Comment by bwateratmsft** [comment]
@Naninani I looked at our code and it appears this warning message was created to work around https://github.com/Azure/azure-cli/issues/4843 -- now fixed for years. I'll take a look to see if it is needed anymore.

**Comment by bwateratmsft** [comment]
@Naninani the problem still exists and is outside of our control--wincred has a limit of 2.5K and the refresh tokens used for CLI login far exceed that. In an [old comment](https://github.com/Azure/azure-cli/issues/4843#issuecomment-342712079), @djyou implied that there would be a server-side change to limit the token length; @djyou, has that actually happened? A refresh token I got just now is 2717 characters long, so the error reproduces.

In the meantime, I'm going to clean up the messy grammar in the message. You are absolutely correct that turning off wincred is less secure, but there's nothing the Docker extension can do about the limitations of it.

**Comment by bwateratmsft** [comment]
We think this should be fixed by #2153. Users who want to remove the `"credsStore": "desktop"` entry can do so, but may need to log out and back in.

**Comment by bwateratmsft** [comment]
We have released Docker 1.4.0 which contains a fix for this.
